workers/common/src/main/kotlin/common/Extensions.kt

```diff
@@ -48,14 +50,45 @@ inline fun <reified T> ConfigManager.readConfigFileValueWithDefault(
     defaultPath: String,
     fallbackValue: T,
     context: Context?
+): T = getConfigFileWithDefault<T>(path, defaultPath, fallbackValue, context, ::readConfigFileValue)
+
+/**
+ * If [path] is not `null`, read the file using the provided [context]. If a [ConfigException] occurs while reading the
+ * file it is rethrown. If [path] is `null`, the file at [defaultPath] is read instead. If the file cannot be read, the
+ * [fallbackValue] is returned.
+ *
+ * This function realizes the contract that if a specific config file is requested, not being able to read it leads to
+ * an exception, but the default file is allowed to not exist.
+ */
+fun ConfigManager.readConfigFileWithDefault(
+    path: String?,
+    defaultPath: String,
+    fallbackValue: String,
+    context: Context?
+): String = getConfigFileWithDefault(path, defaultPath, fallbackValue, context, ::readConfigFile)
```

Comment:
Can't we just get rid of this function by making `::readConfigFile` the default argument for `getConfigFileWithDefault`? Similar for `readConfigFile` below.

When doing do, I'd then rename `getConfigFileWithDefault` to `readConfigFileWithDefault` and `getConfigFile` to `readConfigFile` in order to not also have read vs. get variants of a function, but only "File" vs "FileValue" and optional "WithDefault" variants of the functions.

Reply:
I think the reason I kept the functions separate was that the `get` function returns an input stream like the `getConfigFile` function of the `ConfigManager`. And I wanted to deserialize the input stream directly without converting it to a string first. If you want to optimize something, can you make a follow-up PR?